zebrad/tests/acceptance.rs

```diff
@@ -887,10 +887,22 @@ fn sync_until(
     if enable_mempool_at_height.is_some() {
         child.expect_stdout_line_matches("enabling mempool for debugging")?;
         child.expect_stdout_line_matches("activating mempool")?;
+
+        // make sure zebra is running with the mempool
+        child.expect_stdout_line_matches("verified checkpoint range block_count=400")?;
     }
 
     child.expect_stdout_line_matches(stop_regex)?;
-    child.kill()?;
+
+    // make sure there is never a mempool if we don't explicity enable it
+    if enable_mempool_at_height.is_none() {
+        // this call failure will kill the process
+        assert!(child
+            .expect_stdout_line_matches("activating mempool")
+            .is_err());
+    } else {
+        child.kill()?;
+    }
```

Comment:
Future bug risk: What if we add more code here, and forget to kill the child process?

It's ok to try to kill the child process multiple times, or if it has already exited.

```suggestion
    } 
    
    child.kill()?;
```

Reply:
Done this as part of https://github.com/ZcashFoundation/zebra/pull/2880/commits/31f736366ec2dd36c27cea13d35af875ca5cf02c